### PR TITLE
Updating private key plugin

### PR DESCRIPTION
**Pull request by domanchi**
Addressing https://github.com/Yelp/detect-secrets/issues/31, and upgrading the private key plugin.

### CODE REVIEW

File: detect_secrets/plugins/private_key.py [hunk 1]
```diff
@@ -10,14 +36,14 @@
     'BEGIN EC PRIVATE KEY',
     'BEGIN OPENSSH PRIVATE KEY',
     'BEGIN PRIVATE KEY',
+    'PuTTY-User-Key-File-2',
+    'BEGIN SSH2 ENCRYPTED PRIVATE KEY',
 )
 
 
 class PrivateKeyDetector(BasePlugin):
```
Comment (KevinHock) [inline detect_secrets/plugins/private_key.py]: GitHub will only let me comment on the diff, however so we don't forget what asottile had brought up, shall we do `s/file.readline/file.read/g` on https://github.com/Yelp/detect-secrets/pull/35/files#diff-e2fb9e2d9b26d21395825207c6fb0cd2R63

Comment (domanchi) [reply detect_secrets/plugins/private_key.py]: It's by design that we only check the first line.

https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/plugins/private_key.py#L26

Comment (asottile) [reply detect_secrets/plugins/private_key.py]: hmm, I'd be concerned about false-negatives here, someone could embed a private key in a string and not get caught?

Comment (domanchi) [reply detect_secrets/plugins/private_key.py]: In investigations, looks like this issue is more prominent than I originally thought.

Will update.

**Pull request merged**